src/error.ts

```diff
@@ -705,41 +742,57 @@ const RETRYABLE_ERROR_CODES = new Set<number>([
   MONGODB_ERROR_CODES.NotPrimaryOrSecondary
 ]);
 
+// see: https://github.com/mongodb/specifications/blob/master/source/retryable-writes/retryable-writes.rst#terms
 const RETRYABLE_WRITE_ERROR_CODES = new Set<number>([
-  MONGODB_ERROR_CODES.InterruptedAtShutdown,
-  MONGODB_ERROR_CODES.InterruptedDueToReplStateChange,
-  MONGODB_ERROR_CODES.NotWritablePrimary,
-  MONGODB_ERROR_CODES.NotPrimaryNoSecondaryOk,
-  MONGODB_ERROR_CODES.NotPrimaryOrSecondary,
-  MONGODB_ERROR_CODES.PrimarySteppedDown,
-  MONGODB_ERROR_CODES.ShutdownInProgress,
-  MONGODB_ERROR_CODES.HostNotFound,
-  MONGODB_ERROR_CODES.HostUnreachable,
-  MONGODB_ERROR_CODES.NetworkTimeout,
-  MONGODB_ERROR_CODES.SocketException,
+  ...RETRYABLE_READS_ERROR_CODES,
   MONGODB_ERROR_CODES.ExceededTimeLimit
 ]);
 
-export function isRetryableEndTransactionError(error: MongoError): boolean {
-  return error.hasErrorLabel('RetryableWriteError');
-}
+export function isRetryableWriteError(error: MongoError, maxWireVersion: number): boolean {
+  if (!(error instanceof MongoError)) {
```

Comment:
thought: What if we put a static method on the `MongoError` class to determine whether something was a `MongoError`?  We could add a type assertion too to help Typescript, if we ever cared about that

```typescript
class MongoError {
    ...

    static isMongoError(err: any) : err is MongoError { error instanceof MongoError }
}
```

Reply:
I think it's a good idea! While fixing this I got some real power out of our many specific errors, I'm wondering if we can do something that's generic, and available on all error classes as an improvement

Reply:
I think we should be careful about double wrapping things in functions that already have straightforward assertions (like `instanceof`), for performance reasons; I like the basic idea for readability and such, I just think we should consider it more carefully